packages/promise/src/PromiseCache.ts

```diff
@@ -0,0 +1,141 @@
+import type { Key, SuspensePromiseOptions } from './types'
+import { hashKey } from './utils'
+
+type Sync = (...args: unknown[]) => unknown
+
+type PromiseCacheState<TKey extends Key = Key> = {
+  promise?: Promise<unknown>
+  promiseKey: TKey
+  hashedKey: ReturnType<typeof hashKey>
+  error?: unknown
+  data?: unknown
+}
+
+/**
+ * @experimental This is experimental feature.
+ */
+export class PromiseCache {
+  private cache = new Map<ReturnType<typeof hashKey>, PromiseCacheState>()
+  private syncsMap = new Map<ReturnType<typeof hashKey>, Sync[]>()
+
+  public reset = (key?: Key) => {
+    if (key === undefined || key.length === 0) {
+      this.cache.clear()
+      this.syncSubscribers()
+      return
+    }
+
+    const hashedKey = hashKey(key)
+
+    if (this.cache.has(hashedKey)) {
+      // TODO: reset with key index hierarchy
+      this.cache.delete(hashedKey)
+    }
+    this.syncSubscribers(key)
+  }
+
+  public clearError = (key?: Key) => {
+    if (key === undefined || key.length === 0) {
+      this.cache.forEach((value, key, map) => {
+        map.set(key, { ...value, promise: undefined, error: undefined })
+      })
+      return
+    }
+
+    const hashedKey = hashKey(key)
+    const promiseCacheState = this.cache.get(hashedKey)
+    if (promiseCacheState) {
+      // TODO: clearError with key index hierarchy
+      this.cache.set(hashedKey, { ...promiseCacheState, promise: undefined, error: undefined })
+    }
+  }
+
+  public invalidate(key: Key) {
+    const hashedKey = hashKey(key)
+
+    if (this.cache.has(hashedKey)) {
+      this.cache.delete(hashedKey)
+    }
+    this.syncSubscriber(key)
+  }
```

Comment:
Invalidate method added to PromiseCache.

Reply:
https://github.com/user-attachments/assets/ef5b9e79-37a8-4d11-8617-16fb6258ec08

Reply:
## Interface

```
const options = promiseOptions({ promsieKey: ["post", 1], promiseFn: () => Promise.resolve("post data") })

promiseCache.invalidate(options.key)

or

primseCache.invalidate(["post", 1])
```

If you have any better ideas for the interface, please comment!